cmd/nodecmd/join.go

```diff
@@ -0,0 +1,478 @@
+		return false, err
+	}
+	isBootstrapped, err := parseBootstrappedOutput(app.GetBootstrappedJSONFile())
+	if err != nil {
+		return false, err
+	}
+	if err := app.RemoveAnsibleStatusDir(); err != nil {
+		return false, err
+	}
+	if isBootstrapped {
+		return true, nil
+	}
+	return false, nil
+}
+
+func getClusterNodeID(clusterName string) (string, error) {
+	ux.Logger.PrintToUser("Getting node id ...")
+	if err := app.CreateAnsibleStatusFile(app.GetNodeIDJSONFile()); err != nil {
+		return "", err
+	}
+	if err := ansible.RunAnsiblePlaybookGetNodeID(app.GetAnsibleDir(), app.GetNodeIDJSONFile(), app.GetAnsibleInventoryPath(clusterName)); err != nil {
+		return "", err
+	}
+	nodeID, err := parseNodeIDOutput(app.GetNodeIDJSONFile())
+	if err != nil {
+		return "", err
+	}
+	if err = app.RemoveAnsibleStatusDir(); err != nil {
+		return "", err
+	}
+	return nodeID, err
+}
+
+func getClusterSubnetSyncStatus(blockchainID, clusterName string) (bool, error) {
+	ux.Logger.PrintToUser("Checking if node is synced to subnet ...")
+	if err := app.CreateAnsibleStatusFile(app.GetSubnetSyncJSONFile()); err != nil {
+		return false, err
+	}
+	if err := ansible.RunAnsiblePlaybookSubnetSyncStatus(app.GetAnsibleDir(), app.GetSubnetSyncJSONFile(), blockchainID, app.GetAnsibleInventoryPath(clusterName)); err != nil {
+		return false, err
+	}
+	subnetSyncStatus, err := parseSubnetSyncOutput(app.GetSubnetSyncJSONFile())
+	if err != nil {
+		return false, err
+	}
+	if err = app.RemoveAnsibleStatusDir(); err != nil {
+		return false, err
+	}
+	if subnetSyncStatus == status.Syncing.String() {
+		return true, nil
+	} else if subnetSyncStatus == status.Validating.String() {
+		return false, errors.New("node is already a subnet validator")
+	}
+	return false, nil
+}
+
+// checkNodeIsPrimaryNetworkValidator only returns err if node is already a Primary Network validator
+func checkNodeIsPrimaryNetworkValidator(nodeID ids.NodeID, network models.Network) error {
+	isValidator, err := subnet.IsSubnetValidator(ids.Empty, nodeID, network)
+	if err != nil {
+		ux.Logger.PrintToUser("failed to check if node is a validator on Primary Network: %s", err)
+	} else if isValidator {
+		return fmt.Errorf("node %s is already a validator on Primary Network", nodeID)
+	}
+	return nil
+}
+
+func addNodeAsPrimaryNetworkValidator(nodeID ids.NodeID, network models.Network) (bool, error) {
+	if err := checkNodeIsPrimaryNetworkValidator(nodeID, network); err == nil {
+		err = validatePrimaryNetwork(nodeID, network)
+		if err != nil {
+			return false, err
+		}
+		ux.Logger.PrintToUser("Node successfully added as Primary Network validator!")
+		return true, nil
+	}
+	return false, nil
+}
+
+func waitForNodeToBePrimaryNetworkValidator(nodeID ids.NodeID) {
+	ux.Logger.PrintToUser("Waiting for the node to start as a Primary Network Validator...")
+	// wait for 20 seconds because we set the start time to be in 20 seconds
```

Comment:
I thought on 229 the start time was 1 minute?

Reply:
Yup I suggested so for ux (if not the user waits a minute here) but either case I just defer to @dasconnor opinion here for the different validator lead times

Reply:
As long as it's consistent, it's fine